### ISSUE TITLE
[[coro_lifetimebound]] does not catch lifetime issues with lambda captures

### ISSUE THREAD

**Issue by usx95** [llvm/llvm-project#76995]
The pattern of interest is 
```cpp
co_task<int> coro() {
    int a = 1;
    auto lamb = [a](int x, const int& y) -> co_task<int> {
        co_return x + y + a; // 'a' in the lambda object dies after the iniital_suspend.
    }(1, a);
    co_return co_await lamb;
}
```
https://godbolt.org/z/GWPEovWWc

Lambda captures (even by value) are prone to use after free once the lambda object dies. In the above example, the lambda object appears only as a temporary in the call expression.

It should be fine if this is used in a `co_await` expression due to lifetime extension of temporaries. Example:

```cpp
co_task<int> coro() {
    int a = 1;
    int res = co_await [a](int x, const int& y) -> co_task<int> {
        co_return x + y + a;
    }(1, a);
    co_return res;
}
```
